crates/biome_js_analyze/tests/specs/nursery/noUselessConcat/invalid.js.snap

```diff
@@ -0,0 +1,398 @@
+
+  ! Useless string concatenation.
+  
+    1 │ const a = 'a' + 'b'
+  > 2 │ const a = 'a' + 'b' + 'c'
+      │           ^^^^^^^^^^^^^^^
+    3 │ const a = 'a' + ('b' + 'c')
+    4 │ const a = ('a' + 'b') + 'c'
+  
+  i Consider turning the expression into a single string to improve readability and runtime performance.
+  
+  i Unsafe fix: Remove the useless concatenation
+  
+     1  1 │   const a = 'a' + 'b'
+     2    │ - const·a·=·'a'·+·'b'·+·'c'
+        2 │ + const·a·=·"abc"
+     3  3 │   const a = 'a' + ('b' + 'c')
+     4  4 │   const a = ('a' + 'b') + 'c'
+  
+
+```
+
+```
+invalid.js:3:11 lint/nursery/noUselessConcat  FIXABLE  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
+
+  ! Useless string concatenation.
+  
+    1 │ const a = 'a' + 'b'
+    2 │ const a = 'a' + 'b' + 'c'
+  > 3 │ const a = 'a' + ('b' + 'c')
+      │           ^^^^^^^^^^^^^^^^^
+    4 │ const a = ('a' + 'b') + 'c'
+    5 │ const a = foo + 'a' + 'b' + 'c'
+  
+  i Consider turning the expression into a single string to improve readability and runtime performance.
+  
+  i Unsafe fix: Remove the useless concatenation
+  
+     1  1 │   const a = 'a' + 'b'
+     2  2 │   const a = 'a' + 'b' + 'c'
+     3    │ - const·a·=·'a'·+·('b'·+·'c')
+        3 │ + const·a·=·"abc"
+     4  4 │   const a = ('a' + 'b') + 'c'
+     5  5 │   const a = foo + 'a' + 'b' + 'c'
+  
+
+```
+
+```
+invalid.js:4:11 lint/nursery/noUselessConcat  FIXABLE  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
+
+  ! Useless string concatenation.
+  
+    2 │ const a = 'a' + 'b' + 'c'
+    3 │ const a = 'a' + ('b' + 'c')
+  > 4 │ const a = ('a' + 'b') + 'c'
+      │           ^^^^^^^^^^^^^^^^^
+    5 │ const a = foo + 'a' + 'b' + 'c'
+    6 │ const a = (foo + 'a') + ('b' + 'c')
+  
+  i Consider turning the expression into a single string to improve readability and runtime performance.
+  
+  i Unsafe fix: Remove the useless concatenation
+  
+     2  2 │   const a = 'a' + 'b' + 'c'
+     3  3 │   const a = 'a' + ('b' + 'c')
+     4    │ - const·a·=·('a'·+·'b')·+·'c'
+        4 │ + const·a·=·"abc"
+     5  5 │   const a = foo + 'a' + 'b' + 'c'
+     6  6 │   const a = (foo + 'a') + ('b' + 'c')
+  
+
+```
+
+```
+invalid.js:5:11 lint/nursery/noUselessConcat  FIXABLE  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
+
+  ! Useless string concatenation.
+  
+    3 │ const a = 'a' + ('b' + 'c')
+    4 │ const a = ('a' + 'b') + 'c'
+  > 5 │ const a = foo + 'a' + 'b' + 'c'
+      │           ^^^^^^^^^^^^^^^^^^^^^
+    6 │ const a = (foo + 'a') + ('b' + 'c')
+    7 │ const a = ((foo + 'a') + ('b' + 'c') + 1)
+  
+  i Consider turning the expression into a single string to improve readability and runtime performance.
+  
+  i Unsafe fix: Remove the useless concatenation
+  
+     3  3 │   const a = 'a' + ('b' + 'c')
+     4  4 │   const a = ('a' + 'b') + 'c'
+     5    │ - const·a·=·foo·+·'a'·+·'b'·+·'c'
+        5 │ + const·a·=·foo·+·'a'·+·"bc"
```

Comment:
I’m not sure I fully understand the reason we can’t merge `'a'` with the other strings. Regardless of the value of `foo`, wouldn’t the end result always end in `"abc"`?

Reply:
I was sure of that too, until I realized that `{} + 'abc'` is different from `{} + 'a' + 'bc'` in browser environments:
Browsers: 
```
{} + 'abc' // NaN
{} + 'a' + 'bc' // "NaNbc"
```

Weirdly enough, the result is totally different in Node: 
```
{} + 'a' + 'bc' // "[object Object]abc"
{} + 'abc' // "[object Object]abc"
```